.github/workflows/ci.yml

```diff
@@ -0,0 +1,136 @@
+name: ci
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+        with:
+          fetch-depth: 0
+
+      - name: Setup Go
+        uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version: "1.19"
+          cache: true
+
+      - name: ci/check-style
+        run: make check-style
+
+      - name: ci/check-modules
+        run: make check-modules
+
+      - name: ci/check-boilerplate
+        run: make check-modules
+
+      - name: ci/check-mocks
+        run: make verify-mocks
```

Comment:
Would suggest in encapsulating this, in a single makefile target
And then ghactions can call something like `make lint` to perform all related tests. 

Thus we decapsulate the specifics of the project completely from the actual CI. To keep the pipeline clean and readable and easily to be imported to other projects.

Reply:
Resolving since it is out of the scope. We are only migrating to ghactions in this iteration. 
Makefile concerns to be addressed in next iteration.